spring-cloud-circuitbreaker-resilience4j/src/main/java/org/springframework/cloud/circuitbreaker/resilience4j/ReactiveResilience4JCircuitBreakerFactory.java

```diff
@@ -33,24 +34,41 @@
 
 /**
  * @author Ryan Baxter
+ * @author Thomas Vitale
  */
 public class ReactiveResilience4JCircuitBreakerFactory extends
 		ReactiveCircuitBreakerFactory<Resilience4JConfigBuilder.Resilience4JCircuitBreakerConfiguration, Resilience4JConfigBuilder> {
 
-	private Function<String, Resilience4JConfigBuilder.Resilience4JCircuitBreakerConfiguration> defaultConfiguration = id -> new Resilience4JConfigBuilder(
-			id).circuitBreakerConfig(CircuitBreakerConfig.ofDefaults())
-					.timeLimiterConfig(TimeLimiterConfig.ofDefaults()).build();
+	private Function<String, Resilience4JConfigBuilder.Resilience4JCircuitBreakerConfiguration> defaultConfiguration;
 
 	private CircuitBreakerRegistry circuitBreakerRegistry = CircuitBreakerRegistry.ofDefaults();
 
+	private TimeLimiterRegistry timeLimiterRegistry = TimeLimiterRegistry.ofDefaults();
+
 	private Map<String, Customizer<CircuitBreaker>> circuitBreakerCustomizers = new HashMap<>();
 
+	@Deprecated
+	public ReactiveResilience4JCircuitBreakerFactory() {
+		this.defaultConfiguration = id -> new Resilience4JConfigBuilder(id)
+				.circuitBreakerConfig(CircuitBreakerConfig.ofDefaults())
+				.timeLimiterConfig(TimeLimiterConfig.ofDefaults()).build();
+	}
+
+	public ReactiveResilience4JCircuitBreakerFactory(CircuitBreakerRegistry circuitBreakerRegistry,
+			TimeLimiterRegistry timeLimiterRegistry) {
+		this.circuitBreakerRegistry = circuitBreakerRegistry;
+		this.timeLimiterRegistry = timeLimiterRegistry;
+		this.defaultConfiguration = id -> new Resilience4JConfigBuilder(id)
+				.circuitBreakerConfig(this.circuitBreakerRegistry.getDefaultConfig())
+				.timeLimiterConfig(this.timeLimiterRegistry.getDefaultConfig()).build();
```

Comment:
I don't think this maps `instance` specific properties. Should it be:
```suggestion
		this.defaultConfiguration = id -> {
		    Optional<CircuitBreakerConfig> instanceCircuitBreakerConfig = this.circuitBreakerRegistry.getConfiguration(id);
          CircuitBreakerConfig circuitBreakerConfig = instanceCircuitBreakerConfig.isPresent()
              ? instanceCircuitBreakerConfig.get()
              : this.circuitBreakerRegistry.getDefaultConfig();
          Optional<TimeLimiterConfig> instanceTimeLimiterConfig = this.timeLimiterRegistry.getConfiguration(id);
          TimeLimiterConfig timeLimiterConfig = instanceTimeLimiterConfig.isPresent()
              ? instanceTimeLimiterConfig.get()
              : this.timeLimiterRegistry.getDefaultConfig()
		    return new Resilience4JConfigBuilder(id)
				.circuitBreakerConfig(circuitBreakerConfig)
				.timeLimiterConfig(timeLimiterConfig).build();
		}
```

Reply:
Thanks for your comment. It doesn't include the `instance` specific configuration because `this.defaultConfiguration` contains only the default one. When calling `ReactiveResilience4JCircuitBreakerFactory#create()`, the implementation will use the `instance` configuration (if it exists) or else it falls back on the default one.

Example when instance config is defined, so the default is not used: https://github.com/spring-cloud/spring-cloud-circuitbreaker/pull/108/files#diff-ea2aaf2b4d012d6085e070a49f15d846c694e508219f79d5892e4005b4acd7cbR48

Example when instance config is not defined, so the default is used: https://github.com/spring-cloud/spring-cloud-circuitbreaker/pull/108/files#diff-ea2aaf2b4d012d6085e070a49f15d846c694e508219f79d5892e4005b4acd7cbR67